inception_reports/generate_reports_manager.py

```diff
@@ -56,178 +56,516 @@
-    )
-
-    ax2 = plt.subplot(gs[2])
-    colors = plt.cm.tab10(range(len(type_counts.keys())))
-    ax2.set_title("Types of Annotations")
-    ax2.barh(
-        [f"{type} = {value}" for type, value in type_counts.items()],
-        list(type_counts.values()),
-        color=colors,
-    )
-    ax2.set_xlabel("Number of Annotations")
-    
-    fig.suptitle(
-        f'{project_name.split(".")[0]}\nTotal Annotations: {total_annotations}',
-        fontsize=16,
-    )
-    fig.tight_layout()
-    st.pyplot()
-
-    export_data(project_data)
-
-
-def find_element_by_name(element_list, name):
+    Creates a directory in the user's home directory for storing Inception reports imported over the API.
     """
-    Finds an element in the given element list by its name.
+    home_dir = os.path.expanduser("~")
+    new_dir_path = os.path.join(home_dir, ".inception_reports")
+    try:
+        os.makedirs(new_dir_path)
+        os.makedirs(os.path.join(new_dir_path, "projects"))
+    except FileExistsError:
+        pass
 
-    Args:
-        element_list (list): A list of elements to search through.
-        name (str): The name of the element to find.
 
-    Returns:
-        str: The UI name of the found element, or the last part of the name if not found.
-    """
-    for element in element_list:
-        if element.name == name:
-            return element.uiName
-    return name.split(".")[-1]
+def set_sidebar_state(value):
+    if st.session_state.sidebar_state == value:
+        st.session_state.flag = value
+        st.session_state.sidebar_state = (
+            "expanded" if value == "collapsed" else "collapsed"
+        )
+    else:
+        st.session_state.sidebar_state = value
+    st.rerun()
 
 
-def get_type_counts(annotations):
+def translate_tag(tag):
     """
-    Calculate the count of each type in the given annotations. Each annotation is a CAS object.
-
-    Args:
-        annotations (dict): A dictionary containing the annotations.
-
-    Returns:
-        dict: A dictionary containing the count of each type.
+    Translate the given tag to a human-readable format.
     """
-    count_dict = {}
 
-    layerDefinition = annotations.popitem()[1].select(
-        "de.tudarmstadt.ukp.clarin.webanno.api.type.LayerDefinition"
-    )
-    for doc in annotations:
-        type_names = [
-            (
-                find_element_by_name(layerDefinition, t.name),
-                len(annotations[doc].select(t.name)),
-            )
-            for t in annotations[doc].typesystem.get_types()
-            if t.name != "de.tudarmstadt.ukp.clarin.webanno.api.type.LayerDefinition"
-        ]
-
-        for type_name, count in type_names:
-            if type_name not in count_dict:
-                count_dict[type_name] = 0
-            count_dict[type_name] += count
-    count_dict = {k: v for k, v in count_dict.items() if v > 0}
-    count_dict = dict(sorted(count_dict.items(), key=lambda item: item[1]))
+    specialites = {
```

Comment:
We would like the reporting tool to work generically - not only for GeMTeX. Finding places where specific settings or mappings can be customized is part of that. It would be great if these mappings could be pulled out of the code into a configuration file (maybe CSV?, or JSON, YAML?).

If you see this as a kind of translation, maybe a i18n file?